### PR TITLE
feat: 604 adaptation extend analysis config dataclass and reader

### PR DESCRIPTION
**Pull request by ArdtK**
## Issue addressed
Solves #604 

## Code of conduct
- [x] I HAVE NOT added sensitive or compromised (test) data to the repository.
- [x] I HAVE NOT added vulnerabilities to the repository.
- [x] I HAVE discussed my solution with (other) members of the RA2CE team.

## What has been done?
Add a new section for adaptation in the `AnalysisConfigSection`, including a section per adaptation option.

### Checklist
- [x] Code is formatted using our custom `black` and `isort` definitions.
- [x] Tests are either added or updated.
- [x] Branch is up to date with `master`.
- [ ] Updated documentation if needed.

## Additional Notes (optional)
Added `ADAPTATION` to `AnalysisEnum`, to be merged with `AnalysisDamagesEnum` (also containing 1 valid value) in a next change.

### CODE REVIEW

File: ra2ce/analysis/analysis_config_data/analysis_config_data.py [hunk 1]
```diff
@@ -157,6 +157,41 @@ class AnalysisSectionDamages(AnalysisSectionBase):
...
 
+@dataclass
+class AnalysisSectionAdaptation(AnalysisSectionBase):
+    """
+    Reflects all possible settings that an adaptation analysis section might contain.
+    """
+
+    analysis: str = ""
+    discount_rate: float = 0.0
+    time_horizon: float = 0.0
```
Comment (MatthiasHauthDeltares) [inline ra2ce/analysis/analysis_config_data/analysis_config_data.py:168]: After discussion with Chamidu, `time_horizon` should be an argument of `AnalysisSectionAdaptationOption`

File: ra2ce/analysis/analysis_config_data/analysis_config_data_reader.py [hunk 2]
```diff
@@ -296,6 +324,8 @@ def get_analysis_sections(self) -> list[AnalysisSectionBase]:
                 _analysis_section = self._get_analysis_section_damages(_section_name)
             elif _analysis_type in LossesAnalysisNameList:
                 _analysis_section = self._get_analysis_section_losses(_section_name)
+            elif _analysis_type == "adaptation":
```
Comment (MatthiasHauthDeltares) [inline ra2ce/analysis/analysis_config_data/analysis_config_data_reader.py:327]: is it okay to have this as a string?

**Review by MatthiasHauthDeltares — COMMENTED** [review]
LGTM, i like no_adaptation better than no_intervention :)

File: tests/test_data/acceptance_test_data/analyses.ini [hunk 3]
```diff
@@ -47,4 +47,33 @@ weighing = distance
...
+analysis = adaptation
+discount_rate = 0.05
+time_horizon = 10
+vat = 0.21
+climate_factor = 0.2
+initial_frequency = 0.2
+save_gpkg = True
+save_csv = True
+
+[adaptationoption0]
```
[the remark this answers is not in the document]
Comment (Carsopre) [reply tests/test_data/acceptance_test_data/analyses.ini:63]: Or add a property `analysis = analysis7`

File: tests/analysis/analysis_config_data/test_analysis_config_data.py [hunk 4]
```diff
@@ -59,6 +70,22 @@ def test_damages(self, valid_config: AnalysisConfigData):
         # 3. Verify expectations
         assert all(item in _damages for item in DamagesAnalysisNameList)
 
+    def test_adaptation(self, valid_config: AnalysisConfigData):
+        # 1. Define test data
+
+        # 2. Run test
+        _adaptation = valid_config.adaptation
```
Comment (Carsopre) [inline tests/analysis/analysis_config_data/test_analysis_config_data.py:77]:
```suggestion
        # 1. Define test data and run test
        _adaptation = valid_config.adaptation
```

**Review by Carsopre — COMMENTED** [review]
Minor remarks, mostly related to requirement(s) definition.

**Pull request merged**